src/Controls/src/SourceGen/RouteSyntaxReceiver.cs

```diff
@@ -0,0 +1,267 @@
+{
+	public class RouteSyntaxReceiver : ISyntaxReceiver
+	{
+		const string RouteAttribute = "Microsoft.Maui.Controls.RouteAttribute";
+		const string MauiServiceAttribute = "Microsoft.Maui.Controls.MauiServiceAttribute";
+
+		private readonly List<Service> _services = new();
+		private readonly List<RoutedPage> _routedPages = new();
+		private readonly List<ClassDeclarationSyntax> _shellPages = new();
+
+		internal IEnumerable<Service> Services => _services;
+
+		internal IEnumerable<RoutedPage> RoutedPages => _routedPages;
+
+		internal IEnumerable<ClassDeclarationSyntax> ShellPages => _shellPages;
+
+		internal ClassDeclarationSyntax? MauiApp { get; private set; }
+
+		internal ClassDeclarationSyntax? MauiStartup { get; private set; }
+
+		public void OnVisitSyntaxNode(SyntaxNode syntaxNode)
+		{
+			if (syntaxNode is not ClassDeclarationSyntax cds)
+			{
+				return;
+			}
+
+			if (cds.BaseList is not null)
+			{
+				foreach (var baseType in cds.BaseList.Types)
+				{
+					if (baseType.Type is not IdentifierNameSyntax identifierName)
+					{
+						continue;
+					}
+
+					if (identifierName.Identifier.ValueText == "Application")
+					{
+						MauiApp = cds;
+						break;
+					}
+					else if (identifierName.Identifier.ValueText == "Shell")
+					{
+						_shellPages.Add(cds);
+						break;
+					}
+				}
+			}
+
+			if (cds.Identifier.ValueText == "MauiProgram")
+			{
+				foreach (var member in cds.Members)
+				{
+					if (member is MethodDeclarationSyntax { Identifier.ValueText: "CreateMauiApp" })
+					{
+						MauiApp = cds;
+						break;
+					}
+				}
+			}
+			else if (cds.AttributeLists.Count > 0)
+			{
+				IEnumerable<AttributeArgumentSyntax>? arguments;
+
+				var routeAttributes = cds.AttributeLists.SelectMany(al => al.Attributes)
+					.Where(a => a.Name.NormalizeWhitespace().ToFullString().Replace("Attribute", string.Empty) == "Route");
+
+				var routeAttribute = routeAttributes.FirstOrDefault();
+
+				var serviceAttributes = cds.AttributeLists.SelectMany(al => al.Attributes)
+					.Where(a => a.Name.NormalizeWhitespace().ToFullString().Replace("Attribute", string.Empty) == "MauiService");
+
+				var serviceAttribute = serviceAttributes.FirstOrDefault();
+
+				string lifetime;
+
+				if (routeAttribute is not null)
+				{
+					string? viewModelType = null;
+					var implicitViewModel = false;
+					var route = string.Empty;
+					var routes = new List<string>();
+
+					var routeArgument = true;
+					lifetime = Singleton;
+					arguments = routeAttribute.ArgumentList?.Arguments;
+
+					if (arguments is not null)
+					{
+						foreach (var argument in arguments)
+						{
+							if (argument.NameColon is not null)
+							{
+								switch (argument.NameColon.Name.Identifier.Text)
+								{
+									case "route":
+										route = GetRoute(cds, argument.Expression);
+										break;
+									case "lifetime":
+										lifetime = GetLifetime(argument.Expression);
+										break;
+								}
+							}
```

Comment:
This is me trying to simplify the reading go the code, we can do a switch on a 'string?', that means that we could do: 
```suggestion
							
								switch (argument.NameColon?.Name.Identifier.Text)
								{
									case "route":
										route = GetRoute(cds, argument.Expression);
										break;
									case "lifetime":
										lifetime = GetLifetime(argument.Expression);
										break;
								}
							
```
The same can be done with the next switch statement too